File: php/src/StringProblem/LongestCommonPrefix.php
<?php

namespace IbrahimMH\LeetCode\Solution\StringProblem;



class LongestCommonPrefix{


	public function getLongestCommonPrefix(array $list){

		$prefix = null;

		foreach ($list as $word) {
			if (is_null($prefix) || strlen($word) < strlen($prefix)) {
				$prefix = $word;
			}
		}

		foreach($list as $word){
			for ($i=0; $i <= strlen($prefix) - 1; $i++) { 
					if ($word[i] != $prefix[$i]) {
						$prefix = substr($word,0, $i);
					}
			}
		}

		return $prefix;
	}
}

File: python/string_problem/longest_common_prefix.py




def get_common_prefix(words):

	if not words :
		return ''
	
	prefix = words[0]
		
	for word in words[1:]:

		while word[:len(prefix)] != prefix and prefix:
			prefix = prefix[:len(prefix) - 1];


		
	if not prefix :
		
		return ''

	return prefix;		